website/project/views/drafts.py

```diff
@@ -0,0 +1,282 @@
+from flask import request, redirect
+import httplib as http
+from dateutil.parser import parse as parse_date
+import functools
+
+from modularodm import Q
+from modularodm.exceptions import ValidationValueError
+
+from framework.mongo import database
+from framework.mongo.utils import get_or_http_error, autoload
+from framework.exceptions import HTTPError
+from framework.status import push_status_message
+
+from website.exceptions import NodeStateError
+from website.util.permissions import ADMIN
+from website.project.decorators import (
+    must_be_valid_project,
+    must_have_permission,
+    http_error_if_disk_saving_mode
+)
+from website import language
+from website.project import utils as project_utils
+from website.project.model import MetaSchema, DraftRegistration, DraftRegistrationApproval
+from website.project.metadata.utils import serialize_meta_schema, serialize_draft_registration
+from website.project.utils import serialize_node
+
+get_schema_or_fail = lambda query: get_or_http_error(MetaSchema, query)
+autoload_draft = functools.partial(autoload, DraftRegistration, 'draft_id', 'draft')
+
+@autoload_draft
+@must_have_permission(ADMIN)
+@must_be_valid_project
+def submit_draft_for_review(auth, node, draft, *args, **kwargs):
+    """Submit for approvals and/or notifications
+
+    :return: serialized registration
+    :rtype: dict
+    """
+    # TODO(lyndsysimon): Is this view is called for all schemas?
+    approval = DraftRegistrationApproval(
+        initiated_by=auth.user,
+        end_date=None,
+    )
+    approval.save()
+    draft.approval = approval
+    draft.approval.ask(node.active_contributors())
+    draft.save()
+
+    ret = project_utils.serialize_node(node, auth)
+    ret['success'] = True
+    return ret
+
+@autoload_draft
+@must_have_permission(ADMIN)
+@must_be_valid_project
+def draft_before_register_page(auth, node, draft, *args, **kwargs):
+    """Allow the user to select an embargo period and confirm registration
+
+    :return: serialized Node + DraftRegistration
+    :rtype: dict
+    """
+    ret = serialize_node(node, auth, primary=True)
+
+    ret['draft'] = serialize_draft_registration(draft, auth)
+    return ret
+
+@autoload_draft
+@must_have_permission(ADMIN)
+@must_be_valid_project
+@http_error_if_disk_saving_mode
+def register_draft_registration(auth, node, draft, *args, **kwargs):
+    """Initiate a registration from a draft registration
+
+
+    :return: success message; url to registrations page
+    :rtype: dict
+    """
+
+    data = request.get_json()
+    register = draft.register(auth)
+    draft.save()
```

Comment:
It would be helpful to go through the sequence of events in this function together.

1. It seems that the draft is registered (and saved) before certain registration options are validated / set. Although `register.save` is not called yet, the draft (and potentially certain DB references) will have been saved. Will this cause a problem?

2. I'd like to verify that both the `embargo_registration` and `require_approval` branches are requesting confirmation from authorizers as appropriate. (L87 vs L 92)